csharp_dotnetcore/18.Multi-Lingual-Bot/readme.md

```diff
@@ -0,0 +1,64 @@
+The middleware is driven by user state. This means that users can specify their language preference, and the middleware automatically will intercept messages back and forth and present them to the user in their preferred language.
+Users can change their language preference anytime, and since this gets written to the user state, the middleware will read this state and instantly modify its behavior to honor the newly selected preferred langugage.
+
+The [Microsoft Translator Text API](https://docs.microsoft.com/en-us/azure/cognitive-services/translator/), Microsoft Translator Text API is a cloud-based machine translation service. With this API you can translate text in near real-time from any app or service through a simple REST API call. 
+The API uses the most modern neural machine translation technology, as well as offering statistical machine translation technology.
+
+## Overview
+
+In this sample, we create a simple bot that prompts user for their preferred language, and stores the user's preferred language selection in the user state. 
+We also create a middleware that reads user preferred language, and if it is different from the default language (English), calls the Microsoft Translator Text API to translate to and from the user's preferred langugage.
+This means that the bot always receives utterances in English, while users writes and gets responses in their selected language.
+
+Note that this is a very simple example, but shows very powerful principles. 
+The translation middleware allows us to intercept and traslate messages, and the user preferences stored in the user state at the application level allows us to influence the middleware behavior.
+A more ellaborated next step would be to use the Microsoft Translator Text API to detect language, and if the user changes language, automatically switch to that language, without explicitly prompting but running detection on every step.
+
+# To try this sample
+
+- Clone the samples repository
+```bash
+git clone https://github.com/Microsoft/botbuilder-samples.git
+```
+
+## Microsoft Translator Text API
+
+To consume the Microsoft Translator Text API, first obtain a key following the instructions in the [Microsoft Translator Text API documentation](https://docs.microsoft.com/en-us/azure/cognitive-services/translator/translator-text-how-to-signup). 
+Paste the key in the ```translationKey``` placeholder within the appsettings.json file.
+
+## Install BotBuilder tools
+
+- In a terminal, navigate to the samples folder (`BotBuilder-Samples\csharp_dotnetcore\18.Multi-Lingual-Bot`) 
+
+    ```bash
+    cd BotBuilder-Samples\csharp_dotnetcore\18.Multi-Lingual-Bot
+    ```
+
+## Visual Studio
+- Navigate to the samples folder (`BotBuilder-Samples\csharp_dotnetcore\18.Multi-Lingual-Bot`) and open MessageRoutingBot.csproj in Visual Studio 
```

Comment:
>MessageRoutingBot [](start = 104, length = 17)

MessageRoutingBot.csproj -> MultiLingualBot.csproj

Reply:
This still seems to be here. 

---
In reply to: [216844701](https://github.com/Microsoft/BotBuilder-Samples/pull/356#discussion_r216844701) [](ancestors = 216844701)